frontend/src/app/home/components/menu/search/search.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { map, startWith } from 'rxjs/operators';
+import { MainService } from '../../../services/main.service'
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+    selector: 'app-search',
+    templateUrl: './search.component.html',
+    styleUrls: ['./search.component.scss']
+})
+export class SearchComponent implements OnInit {
+    searchControl = new FormControl();
+    filteredOptions: Observable<string[]>;
+
+    constructor(
+        private mainService: MainService,
+        private _snackBar: MatSnackBar) {
+    }
+
+    ngOnInit() {
+        this.mainService.options = this.mainService.kitchen.map(x => x.name);
```

Comment:
jeżeli już bezpośrednio wykorzystujemy prope z serwisu to nie powinniśmy ich edytować bo to może prowadzić do nieoczekiwanych zachowań.